qiskit_experiments/library/randomized_benchmarking/interleaved_rb_experiment.py

```diff
@@ -97,6 +140,14 @@ def __init__(
             seed=seed,
             full_sampling=full_sampling,
         )
+        # Convert interleaved element to integer for speed in 1Q or 2Q case
+        if self.num_qubits == 1:
+            self._interleaved_elem = num_from_1q_circuit(interleaved_clifford.to_circuit())
+        elif self.num_qubits == 2:
+            self._interleaved_elem = num_from_2q_circuit(interleaved_clifford.to_circuit())
+        else:
+            self._interleaved_elem = interleaved_clifford
+        self._interleaved_op = interleaved_element
```

Comment:
I'm finding the handling of the interleaved element a bit confusing, both the names, and the fact that the value of `_interleaved_op` depends on whether circuits has been called yet. What do you think about the following modifications:

1. rename `_interleaved_elem` to `_interleaved_cliff`  because its value is always a clifford (or number representing clifford)
2. never modify the value of `_interleaved_op` past construction
3. introduce a third variable `_interleaved_instruction` which `__set_up_interleaved_op()` (renamed to `__setup_interleaved_instruction()`) updates whenever called
4. Don't call `__setup_interleaved_instruction()` during `circuits()`. instead, overload `_set_backend()` and call it there, because that's the only time when the value of `_interleaved_instruction` needs to be updated

Please let me know what you think.

Reply:
I have no objection to rename `_interleaved_elem` and `_interleaved_op` to `_interleaved_cliff` and `_interleaved_instruction` respectively.
I think your suggestion on caching interleaved instruction would be the best if we had a different design in experiments framework. However, I think it would be safer to set `__interleaved_instruction` every time `circuits` called, given the current design of experiments framework that allows to change experimental configurations via `set_backend`, `set_options` or `set_transpile_options`. I'm not confident that we can continue to cover all the cases when we need to reset None over the long term. Also computing of `__interleaved_instruction` would not become a bottleneck. IMO, RB module could forbid mutation after construction for backend and transpile options in some future, but that would beyond the scope of this PR. Am I too concerned about edge cases?

Reply:
Thanks for the detailed response. No, I don't think you're too concerned. I think you are right that moving the calculation of the transpiled interleaf is out of scope of this PR. If you think the new names are better, please go ahead and rename them, otherwise, please leave them as they are.

Reply:
OK. Considering your suggestion, I've settled on the followings (in 06f6ee4).
- Rename `_interleaved_elem` to `_interleaved_cliff`.
- Keep `_interleaved_op` as Terra is now shifting to use the term "operation" instead of "instruction" when referring an instruction without (qubit) operands.
- Add `_interleaved_element` just to store the original `interleaved_element` given to the constructor.